shipyard/src/main/java/org/jclouds/shipyard/domain/accounts/CreateAccount.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.shipyard.domain.accounts;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import org.jclouds.json.SerializedNames;
+import org.jclouds.shipyard.domain.roles.RoleInfo;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class CreateAccount {
+
+   public abstract String username();
+
+   public abstract String password();
+   
+   public abstract RoleInfo role();
+   
+   CreateAccount() {
+   }
+
+   @SerializedNames({ "username", "password", "role"})
+   public static CreateAccount create(String username, String password, RoleInfo role) {
+      checkArgument(username.length() > 0, "username must not be empty string");
+      checkArgument(password.length() > 0, "password must not be empty string");
+      if (role == null) role = RoleInfo.create("user", null);
+      return new AutoValue_CreateAccount(username, password, role);
+   }
+   
+   @SerializedNames({ "username", "password", "role"})
```

Comment:
Oh, you're right. It might fail [here](https://github.com/jclouds/jclouds/blob/master/core/src/main/java/org/jclouds/json/internal/NamingStrategies.java#L177-193).

jclouds uses that annotation to get the serialized names for the fields, and matches the declared fields in the class with the annotation values, expecting all fields to have the corresponding value. As having the extra "role" parameter in the annotation looks confusing, can you try just removing the annotation from this method, and leave it only in the factory method that accepts all the parameters? This would be more close to the approach we follow with the constructors, where we only annotate the constructor that has all values.

I haven't tried it with the factory method approach, but looking at the code I think it should work.